### PR TITLE
Rename filter to follow naming convention.

### PR DESCRIPTION
**Pull request by diemtvu**
**What this PR does / why we need it**:

Use `istio.` prefix for filter name.

**Special notes for your reviewer**:

**Release note**:
This PR should be followed with change in [Pilot](https://github.com/istio/istio/blob/master/pilot/pkg/networking/plugin/authn/authentication.go#L42) when updating proxy SHA

### CODE REVIEW

**Comment by diemtvu** [comment]
PTAL

File: src/envoy/http/authn/http_filter_integration_test.cc [hunk 1]
```diff
@@ -104,7 +104,7 @@ TEST_P(AuthenticationFilterIntegrationTest, EmptyPolicy) {
 
 TEST_P(AuthenticationFilterIntegrationTest, SourceMTlsFail) {
   config_helper_.addFilter(R"(
-    name: istio_authn
+    name: istio.authn
     config:
```
Comment (JimmyCYJ) [inline src/envoy/http/authn/http_filter_integration_test.cc]: Is it possible to cover both old name and new name? Once we deprecate the old name, we remove the test as well.

**Comment by diemtvu** [comment]
Actually, it's still possible breaking stuff if we does not update pilot to use the new name right away :(. Making this fully backward compatible is a bit hairy (we will also need to change all filters to fallback to the old name when accessing filter dynamic metadata). On the other hand, change in pilot is just few lines and can be done right after this PR (together with SHA update). @kyessenov , do you think it's ok to do so? Otherwise, I can wait for your PR that merge proxy repo to istio, or not even bother to rename them.

**Comment by kyessenov** [comment]
@diemtvu I am not an owner in either pilot or upgrade parts. You should probably talk to @costinm .

**Comment by diemtvu** [comment]
postpone

**Pull request closed without merging**